packages/playground/src/data-boundary.tsx

```diff
@@ -0,0 +1,44 @@
+import { useStream } from '@final-ui/react'
+import { WebSocketDataSource } from '@final-ui/ws-client'
+import { AlertCircle } from '@tamagui/lucide-icons'
+import React, { PropsWithChildren } from 'react'
+import { H4, SizableText, Text, YStack } from 'tamagui'
+
+// tbd: convert `WebSocketDataSource` to `NetworkDataSource`
+export function DataBoundary({
+  dataSource,
+  path,
+  children,
+}: PropsWithChildren<{ dataSource: WebSocketDataSource; path: string }>) {
+  const { status } = useStream(dataSource.state)
```

Comment:
as discussed, we eventually want to remove the state from the dataSource, and allow the user to pass in the state stream directly into DataBoundary

Reply:
Yeah, I think as a mid-step we can make "state" stream a separate prop, to make it more explicit that it can come from a different place. 

However, I think for "WebSocketDataSource" or other DataSources that are network-related, it is okay if they have an extra "state" prop that is not included in the base DataSource type. 

This is what we have right now.

Reply:
Yep totally fine if some data sources have extra features like the status. I just want to remove it from the main DataSource type

Reply:
Yes, it's not included. This PR, if you look at WebSocket data source, has a different type:
```ts
export type WebSocketDataSource = DataSource & {
  state: DataSourceStateStream
}
```

Reply:
In the future, this will be probably named a higher-level "NetworkDataSource" which is a DataSource with "state". But too soon to speculate that.